crates/router/src/core/routing.rs

```diff
@@ -427,14 +427,20 @@ pub async fn link_routing_config(
                     .unwrap_or_default();
 
             utils::when(
-                dynamic_routing_ref.success_based_dynamic_routing_algorithm_id
-                    == Some(algorithm_id.clone()),
+                matches!(
+                    dynamic_routing_ref.success_based_algorithm,
+                    Some(routing_types::DynamicAlgorithmWithTimestamp {
+                        algorithm_id: Some(_),
```

Comment:
this needs to have `algorithm_id` from request?

```suggestion
                        algorithm_id: Some(algorithm_id),
```